src/sagemaker/amazon/amazon_estimator.py

```diff
@@ -622,13 +633,25 @@ def get_image_uri(region_name, repo_name, repo_version=1):
         if supported_version:
             return get_xgboost_image_uri(region_name, supported_version[0])
 
-        logging.warning(
-            "There is a more up to date SageMaker XGBoost image. "
-            "To use the newer image, please set 'repo_version'="
-            "'%s'. For example:\n"
-            "\tget_image_uri(region, 'xgboost', '%s').",
-            XGBOOST_VERSION_1,
-            XGBOOST_VERSION_1,
-        )
     repo = "{}:{}".format(repo_name, repo_version)
     return "{}/{}".format(registry(region_name, repo_name), repo)
+
+
+def _xgboost_version_compare(repo_version):
+    """Compare xgboost image version with latest version
+
+    Args:
+        repo_version:
+    """
+    if repo_version == 1:
+        return True
+    return _split(XGBOOST_LATEST_VERSION) > _split(repo_version)
```

Comment:
this works for now, but it'd be better to convert the numbers to ints because of this kind of case:

```python
>>> "10" < "9"
True
>>> 10 < 9
False
```

Reply:
This comparison won't work if we have chars in the version, i.e. "1.x-1". It may be simpler to just do a direct string comparison between `XGBOOST_LATEST_VERSION` and the provided `repo_version`.

Reply:
@laurenyu Versions might have characters in them so comparing the tuples instead of ints.

Reply:
@balajitummala right, forgot about that. that still doesn't address the fact that the string comparison with the tuples still has the potential to be inaccurate. I think @aws-patlin's suggestion of doing a direct string comparison is the right approach here:

```python
if repo_version in (1, "latest"):
    return True
return repo_version.startswith(XGBOOST_LATEST_VERSION)
```

Reply:
@laurenyu Modified comparison to just check if exists in the latest versions list